### PR TITLE
[1.2.0] Check for result phase at activate method

### PR DESCRIPTION
**Pull request by chimp1984**
Fixes https://github.com/bisq-network/bisq/issues/3487

### CODE REVIEW

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java [hunk 1]
```diff
@@ -238,7 +239,11 @@ protected void deactivate() {
 
     @Override
     public void onParseBlockCompleteAfterBatchProcessing(Block block) {
-        int chainHeight = daoStateService.getChainHeight();
+        checkForResultPhase(daoStateService.getChainHeight());
+        fillCycleList();
+    }
+
+    private void checkForResultPhase(int chainHeight) {
         if (periodService.getFirstBlockOfPhase(chainHeight, DaoPhase.Phase.RESULT) == chainHeight) {
```
Comment (devinbileck) [inline desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java:247]: Shouldn't this be `<= chainHeight`?

**Review by devinbileck — CHANGES_REQUESTED** [review]
NACK
The vote results are only populated if viewing the results on the first block of the results phase. If you try to view the results after that, they are still empty.

File: desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java (hunk 1 above)
Comment (sqrrm) [reply desktop/src/main/java/bisq/desktop/main/dao/governance/result/VoteResultView.java:247]: `periodService.getFirstBlockOfPhase` gets the height of the first block of the current phase so this check will return true for all `chainHeight` in the `RESULT` phase.

**Review by sqrrm — APPROVED** [review]
NACK

**Comment by ripcurlx** [comment]
@sqrrm I think you approved by mistake -> NACK

**Comment by sqrrm** [comment]
I first approved without testing. Then I read a bit more, checked out and tested and it fails, working on a fix. Don't know how to undo the approve.

**Comment by sqrrm** [comment]
I made a new PR at #3500 that fixes this

**Comment by ripcurlx** [comment]
Superseded by #3500

**Pull request closed without merging**